event-count-monitoring.js

```diff
@@ -0,0 +1,158 @@
+    conditions = eventInstanceService.rqlToCondition(rql, subSelectMap, null, null);
+    var dateArray = new Array();
+    dateArray.push(new Date(dateFromRecent));
+    dateArray.push(new Date());
+    const newquery = services.eventInstanceService.countQuery(conditions, dateArray);
+    const tempTotal = newquery[0].getTotal();
+    var sum=0;
+    tempTotal.keySet().forEach((item) => {
+        sum += tempTotal.get(item);
+    });
+    EVENT_TYPES_SUM.push(sum);
+}
+
+console.log("=============================== NUMBER OF EVENTS FOR RECENT PERIOD =============================== ")
+
+var numberOfEventsForRecentPeriodIndex = 0;
+EVENT_TYPES_SUM.forEach((item) => {
+    // print average events per period
+    console.log(EVENT_TYPES[numberOfEventsForRecentPeriodIndex] + " events for recent period: " + item);
+    numberOfEventsForRecentPeriodIndex++;
+});
+
+
+// Iterate over each event type and get all the event counts
+EVENT_TYPES.forEach(requestTypeRange);
+
+function requestTypeRange(value) {
+
+    let subSelectMap = new Map();
+    var astNodeInitial = new ASTNode("eq", "typeName", value);
+    var rql = RQLUtils.addAndRestriction(astNodeInitial, new ASTNode("ge", "activeTs", dateFromRecentMultiplier.valueOf()));
+    rql = RQLUtils.addAndRestriction(rql, new ASTNode("lt", "activeTs", dateNow.valueOf()));
+    var conditions = new ConditionSortLimit(null, null, null, null);
+    conditions = eventInstanceService.rqlToCondition(rql, subSelectMap, null, null);
+    var dateArray = new Array();
+    dateArray.push(new Date(dateFromRecentMultiplier));
+    dateArray.push(new Date());
+    const newquery = services.eventInstanceService.countQuery(conditions, dateArray);
+    const tempTotal = newquery[0].getTotal();
+
+    var sum=0;
+    tempTotal.keySet().forEach((item) => {
+        sum += tempTotal.get(item);
+    });
+    EVENT_TYPES_RANGE_PERIOD_SUM.push(sum);
+}
+
+console.log("=============================== NUMBER OF TOTAL EVENTS BY TYPE =============================== ")
+
+var numberOfTotalEventsByTypeIndex = 0;
+EVENT_TYPES_RANGE_PERIOD_SUM.forEach((item) => {
+    // print average events per period
+    console.log(EVENT_TYPES[numberOfTotalEventsByTypeIndex] + " events for recent period*multipler by type: " + item);
+    numberOfTotalEventsByTypeIndex++;
+});
+
+
+EVENT_TYPES_RANGE_PERIOD_SUM.forEach((item) => {
+    AVERAGE_EVENTS_PER_PERIOD.push(item/RANGE_PERIOD_MULTIPLE);
+});
+console.log("=============================== NUMBER OF AVERAGE EVENTS PER PERIOD =============================== ")
+
+var eventTypesAverageEventsIndex = 0;
+AVERAGE_EVENTS_PER_PERIOD.forEach((item) => {
+    // print average events per period
+    console.log(EVENT_TYPES[eventTypesAverageEventsIndex] + " average events per period: " + item);
+    eventTypesAverageEventsIndex++;
+});
+
+
+console.log("=============================== EXCEEDS THRESHOLD? =============================== ")
+var exceedsThresholdIndex = 0;
```

Comment:
please use const instead of var

Reply:
Use `let` for variables, use `const` for constants.